src/target/puya.c

```diff
@@ -114,18 +131,32 @@ bool puya_probe(target_s *target)
 	size_t flash_size = 0U;
 
 	const uint32_t dbg_idcode = target_mem32_read32(target, PUYA_DBG_IDCODE);
-	if ((dbg_idcode & 0xfffU) == 0) {
+	const uint16_t dev_id = (dbg_idcode >> PUYA_DBG_IDCODE_DEV_ID_SHIFT) & PUYA_DBG_IDCODE_DEV_ID_MASK;
+	switch (dev_id) {
+	case PUYA_DEV_ID_PY32F002A: {
 		const uint32_t flash_ram_sz = target_mem32_read32(target, PUYA_FLASH_RAM_SZ);
 		flash_size = (((flash_ram_sz >> PUYA_FLASH_SZ_SHIFT) & PUYA_FLASH_SZ_MASK) + 1) << PUYA_FLASH_UNIT_SHIFT;
 		ram_size = (((flash_ram_sz >> PUYA_RAM_SZ_SHIFT) & PUYA_RAM_SZ_MASK) + 1) << PUYA_RAM_UNIT_SHIFT;
-		// TODO: which part families does this actually correspond to?
-		// Tested with a PY32F002AW15U which returns 0x60001000 in IDCODE
-		target->driver = "PY32Fxxx";
-	} else {
+		target->driver = "PY32F002A";
+		break;
+	}
+	case PUYA_DEV_ID_PY32F002B:
+		/*
+		 * 0x1fff0ffc contains 0; did not find any other location that looks like it might contain the flash
+		 * and RAM sizes. We'll hard-code the datasheet values for now. Both flash size and RAM size actually
+		 * match the datasheet value, unlike PY32F002A which (sometimes?) has more RAM and flash than
+		 * documented.
+		 */
+		flash_size = 24 * 1024;
+		ram_size = 3 * 1024;
```

Comment:
Please suffix all these constants with `U` so they are defined unsigned and don't do a signed-unsigned conversion on assignment.

Reply:
Fixed locally; will push together with the other changes once they are done.